pgjdbc/src/main/java/org/postgresql/largeobject/BlobInputStream.java

```diff
@@ -90,8 +90,25 @@ public BlobInputStream(LargeObject lo, int bsize, long limit) {
     // The very first read multiplies the last buffer size by two, so we divide by two to get
     // the first read to be exactly the initial buffer size
     this.lastBufferSize = INITIAL_BUFFER_SIZE / 2;
+
+    long maxBlobSize;
+    try {
+      this.absolutePosition = lo.tell64();
+      maxBlobSize = 4L * 1024 * 1024 * 1024 * 1024 - 2 * 1024;
+    } catch (SQLException e1) {
+      try {
+        this.absolutePosition = lo.tell();
```

Comment:
Document what this is doing ?

Reply:
This line specifically, or the added section?

Reply:
seems to me that we are switching from LO to bytea here ?

Reply:
The whole section. Frankly, the code is non obvious, so it makes sense to document the considered edge cases

Reply:
It is sad we increase the number of roundtrips for a common case of "blob is used only once". Can we avoid adding extra db calls in the constructor?

Reply:
No? On old version of Postgres (which you still test against), large objects only use an int for their position, with a limit of 2GB: https://www.postgresql.org/docs/9.2/lo-intro.html

See also the existing implementation of `reset()`, which checks whether it can use `seek` instead of `seek64`.

Reply:
> Can we avoid adding extra db calls in the constructor?

Not really. Someone might call `mark()` before anything else, and the position needs to be correctly initialised before then.

Reply:
As far as testing goes, if you don't want to test on older versions just disable the test. I'm fine with that.